### PR TITLE
[CWS] add constraints to network context fields

### PR DESCRIPTION
**Pull request by safchain**
### What does this PR do?

This PR adds a mechanism to define constraints to network context field availability. Currently only imds and dns events are allowed to use network fields.

### CODE REVIEW

File: pkg/security/seclwin/model/model.go [hunk 1]
```diff
@@ -311,6 +313,7 @@ type MatchedRule struct {
 // ActionReport defines an action report
 type ActionReport interface {
 	ToJSON() ([]byte, bool, error)
+	IsMatchingRule(ruleID eval.RuleID) bool
```
Comment (paulcacheux) [inline pkg/security/seclwin/model/model.go:316]: This feels unrelated to the PR (and I think it's already on main from another PR)

**Review by paulcacheux — APPROVED** [review]

File: pkg/security/seclwin/model/model.go (hunk 1 above)
Comment (safchain) [reply pkg/security/seclwin/model/model.go:316]: Windows sync

Comment (paulcacheux) [reply pkg/security/seclwin/model/model.go:316]: ah yes sorry ! Great then :)
